### DISCUSSION TITLE
Using gatsby-plugin-google-tagmanager during runtime

### DISCUSSION THREAD

**Discussion by qasimalyas** [gatsbyjs/gatsby#27525] [Help] [answered]
## Summary

I'm using `gatsby-plugin-google-tagmanager` which initially works fine. I can push data using the `defaultDataLayer` option but is there a way to push other data using the plugin from components at runtime? So essentially a function which I can import and use to push data throughout the app?

**Comment by herecydev** [comment, marked as answer]
Yes it should be available via `window.dataLayer` unless you're changing `dataLayerName`.
